plugin/storage/es/spanstore/reader.go

```diff
@@ -205,39 +205,76 @@ func (s *SpanReader) FindTraces(traceQuery *spanstore.TraceQueryParameters) ([]*
 }
 
 func (s *SpanReader) multiRead(traceIDs []string, startTime, endTime time.Time) ([]*model.Trace, error) {
+
 	if len(traceIDs) == 0 {
 		return []*model.Trace{}, nil
 	}
 	searchRequests := make([]*elastic.SearchRequest, len(traceIDs))
 
-	for i, traceID := range traceIDs {
-		query := elastic.NewTermQuery("traceID", traceID)
-		searchRequests[i] = elastic.NewSearchRequest().IgnoreUnavailable(true).Type("span").
-			Source(elastic.NewSearchSource().Query(query).Size(defaultDocCount))
-	}
-
+	var traces []*model.Trace
 	// Add an hour in both directions so that traces that straddle two indexes are retrieved.
-	// ie starts in one and ends in another.
+	// i.e starts in one and ends in another.
 	indices := findIndices(spanIndexPrefix, startTime.Add(-time.Hour), endTime.Add(time.Hour))
 
-	results, err := s.client.MultiSearch().
-		Add(searchRequests...).
-		Index(indices...).
-		Do(s.ctx)
-	if err != nil {
-		return nil, err
-	}
+	var nextTime uint64
```

Comment:
can this be initialized to startTime?

Reply:
I am not sure , because nextTime is derived from the results of the query in line 232.

Reply:
Since nextTime is derived from the results of the previous 10K , don't think we can initialize.

Reply:
doesn't this mean that the first ever query always queries from 0 (searchAfterTime isn't initialized) even though we're only interested things after startTime? (I guess since we're looking at a subset of indices, this isn't too bad performance wise)

Reply:
@black-adder Thought about this a bit... I agree , we should initialize next-time to start-time, don't see a problem there. I am going to modify this and update the PR.